src/color-retriever/index.ts

```diff
@@ -0,0 +1,188 @@
+import { BitMatrix } from "../BitMatrix";
+import { QRLocation } from "../locator";
+import { Point } from "../Point";
+
+// Stores two RGBA values (0-255, [r, g, b, a])
+export interface QRColors {
+  qr: Uint8ClampedArray;
+  background: Uint8ClampedArray;
+}
+
+/* Retrieves the colors that make up a scanned QR code. RGB (assumed to be sRGB) values are converted to the CIELab
+color space for averaging (with no regard for alpha), and then converted back to RGB. Alpha values are simply averaged
+directly. */
+export function retrieveColors(location: QRLocation,
+                               extracted: {matrix: BitMatrix; mappingFunction: (x: number, y: number) => Point; },
```

Comment:
Instead of defining this here, might make sense to define an `Extracted` type in extractor and then reference it here, I feel like that would be a little cleaner per se.